js/ddmlib/ddmclient.js

```diff
@@ -325,7 +325,10 @@ class OfflineServiceController {
     }
 }
 
-class NoOpServiceController {
+class OtioseServiceController {
+    constructor() {
+        this.hasNoImage = true;
```

Comment:
why is this needed?

Reply:
This is useful for hview.js, line 1353:

line 1351:    const showPreviewContext = function(e) {
line 1352:        const menu = [ buildPreviewMenuItem(0, "Grid") ]
line 1353:        if (!viewController.hasNoImage) {
line 1354:             menu.push(null, buildPreviewMenuItem(1, "Image"), null, buildPreviewMenuItem(2, "Both"))
line 1355:         }
line 1355:         menu.push(null, buildPreviewMenuItem(3, "App"))

If a service controller has set the "hasNoImage" boolean, then the preview menu won't show the image or both modes as possible.